qemu/tests/qemu_guest_agent.py

```diff
@@ -1026,6 +1027,124 @@ def gagent_check_with_fsfreeze(self, test, params, env):
 
+    @error_context.context_aware
+    def gagent_check_guest_exec(self, test, params, env):
+        """
+        Execute a command in the guest via guest-exec cmd,
+        and check status of this process.
+
+        Steps:
+        1) Change guest-exec related cmd to white list,linux guest only.
+        2) Execute guest cmd and get the output.
+        3) Check the cmd's result and output from return.
+        4) Execute guest cmd and no need to get the output.
+        5) Check the cmd's result from return.
+        6) Issue an invalid guest cmd.
+        7) Check the return result.
+        8) Execute guest cmd with wrong args.
+        9) Check the return result.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        """
+
+        def _guest_cmd_run(guest_cmd, cmd_args=None, env=None,
+                           input=None, capture_output=None):
+            """
+            Execute guest-exec cmd and get the result in timeout.
+
+            :param guest_cmd: path or executable name to execute
+            :param cmd_args: argument list to pass to executable
+            :param env: environment variables to pass to executable
+            :param input: data to be passed to process stdin (base64 encoded)
+            :param capture_output: bool flag to enable capture of stdout/stderr
+                                   of running process,defaults to false.
+            :return: result of guest-exec cmd
+            """
+
+            # change cmd_args to be a list needed by guest-exec.
+            if cmd_args:
+                cmd_args = cmd_args.split()
+            ret = self.gagent.guest_exec(path=guest_cmd, arg=cmd_args,
+                                         env=env, input_data=input,
+                                         capture_output=capture_output)
+            end_time = time.time() + float(params["guest_cmd_timeout"])
+            while time.time() < end_time:
+                result = self.gagent.guest_exec_status(ret["pid"])
+                if "exited" not in result:
+                    test.fail("The result of guest-exec is not correct.")
+                if result["exited"]:
+                    logging.info("Guest cmd is finished.")
+                    break
+                time.sleep(5)
+
+            if not result["exited"]:
+                test.error("Guest cmd is still running, pls login guest to"
+                           " handle it or extend your timeout.")
+            # check the exitcode and output/error data result
+            if result["exitcode"] == 0:
+                if "out-data" in result:
+                    out_data = base64.b64decode(result["out-data"]).decode()
+                    logging.info("The guest cmd is executed successfully,"
+                                 "the output is: \n %s." % out_data)
+                elif "err-data" in result:
+                    test.fail("When exitcode is 0, should not get error data.")
```

Comment:
Does that really exist? I think result["exitcode"] would be enough to verify the command execution success, just like shell command status, we only need check output when the status is not correct.